packages/components/ovh-shell/src/client/shell-client.ts

```diff
@@ -0,0 +1,52 @@
+import { nanoid } from 'nanoid';
+
+export interface IDeferred {
+  resolve: (value: unknown | PromiseLike<unknown>) => void;
+  reject: (reason?: unknown) => void;
```

Comment:
I'm thinking that maybe a generic would be better suited instead of the unknown type. Something like this : 

`
export interface IDeferred<T> {
   resolve: (value: T | PromiseLike<T>) => void;
  reject: (reason?: T) => void;
}
`
This way, we can specify a type when calling it, instead of inferring it with the unknown, what do you think?

Reply:
since i'm doing this :
```js
    return new Promise((resolve, reject) => {
      this.deferredResponse[uid] = { resolve, reject }; // this is an object implementing IDeferred interface
    });
```
please correct me if i'm wrong but i don't think i would be able to do so, the resolve / reject method signature comes from the Promise type definition